dali/operators/reader/loader/webdataset_loader.cc

```diff
@@ -0,0 +1,437 @@
+  auto& current_sample = wds_shards_metadata_[current_wds_shard_index_][current_sample_index_];
+  current_wds_shard.SeekArchive(current_sample.start_offset);
+
+  vector<char> sample_was_set(sample.size(), false);
+  while (current_wds_shard.TellArchive() < current_sample.end_offset) {
+    DALI_ENFORCE(!current_wds_shard.EndOfArchive(),
+                 make_string("Index file at ", configs_[current_wds_shard_index_],
+                             " reporting a file longer than actual (archive reached an offset ",
+                             std::to_string(current_wds_shard.TellArchive()),
+                             " and the sample is supposed to end at ",
+                             std::to_string(current_sample.end_offset), ")"));
+    // Check in case of encountering a tar entry that is not a file
+    if (current_wds_shard.GetFileType() != detail::TarArchive::ENTRY_FILE) {
+      current_wds_shard.NextFile();
+      continue;
+    }
+
+    // Check in case of encountering an unneeded entry
+    const std::string extension = GetExtension(current_wds_shard.GetFileName());
+    if (!ext_map_.count(extension)) {
+      current_wds_shard.NextFile();
+      continue;
+    }
+
+    // Check in case skipping sample is necessary
+    const std::string source_info = uris_[current_wds_shard_index_] + " at offset " +
+                                    std::to_string(current_wds_shard.TellArchive());
+    if (ShouldSkipImage(source_info)) {
+      MarkCached(sample, sample_was_set, extension, source_info);
+      continue;
+    }
+
+    // Reading the data into the tensors
+    int64_t size = static_cast<int64_t>(current_wds_shard.GetFileSize());
+
+    if (!copy_read_data_) {
+      auto p = current_wds_shard.ReadFile();
+      DALI_ENFORCE(p != nullptr, "Error reading from a file " + uris_[current_wds_shard_index_]);
+      SetDataPointer(sample, sample_was_set, extension, source_info, p, size);
+    } else {
+      uint8_t* shared_tensor_data =
+          ShareDataPointer(sample, sample_was_set, extension, source_info, size);
+      if (shared_tensor_data) {
+        uint64_t n_read = current_wds_shard.Read(static_cast<uint8_t*>(shared_tensor_data), size);
+        DALI_ENFORCE(static_cast<int64_t>(n_read) == size,
+                     "Error reading from a file " + uris_[current_wds_shard_index_]);
+      }
+    }
+
+    current_wds_shard.NextFile();
+  }
+
+  // setting empty components:
+  int not_set_count = 0;
+  for (size_t component_index = 0; component_index < sample.size(); component_index++) {
+    if (!sample_was_set[component_index]) {
+      not_set_count++;
+      sample[component_index].Reset();
+      sample[component_index].set_type(TypeTable::GetTypeInfo(dtypes_[component_index]));
+      sample[component_index].Resize({0});
+    }
+  }
+  if (not_set_count && missing_component_behavior_ != detail::wds::MissingExtBehavior::Empty) {
+    DALI_FAIL(make_string("Index file at ", configs_[current_wds_shard_index_],
```

Comment:
Can we print the line in the config as well?

Reply:
alright

Reply:
Does it provide any value to print offset? Or not really?